scss/_patterns_code-numbered.scss

```diff
@@ -2,43 +2,45 @@
 $sidebar-width: 4.5rem !default;
 
 @mixin vf-p-code-numbered {
-  .p-code-numbered {
-    counter-reset: line-numbering;
-    padding: $sph-inner--small 0;
-  }
-
-  .p-code-numbered__line {
-    display: inline-block;
-    padding: 0 $sph-inner 0 ($sidebar-width + $sph-inner);
-    position: relative;
-    white-space: pre-wrap;
-    width: 100%;
-
-    &:empty {
-      display: block;
-      min-height: $sp-xx-large;
+  @include deprecate('3.0.0', 'p-code-numbered is being removed, use the code snippet block numbered variant instead.') {
+    .p-code-numbered {
+      counter-reset: line-numbering;
+      padding: $sph-inner--small 0;
     }
 
-    &::before {
-      background: transparent;
-      color: $color-mid-dark;
-      content: counter(line-numbering);
-      counter-increment: line-numbering;
+    .p-code-numbered__line {
       display: inline-block;
-      height: 100%;
-      left: 0;
-      padding: 0 $sph-inner;
-      pointer-events: none;
-      position: absolute;
-      text-align: right;
-      top: 0;
-      // stylelint-disable property-no-vendor-prefix
-      -webkit-user-select: none;
-      -moz-user-select: none;
-      -ms-user-select: none;
-      user-select: none;
-      // stylelint-enable property-no-vendor-prefix
-      width: $sidebar-width;
+      padding: 0 $sph-inner 0 ($sidebar-width + $sph-inner);
```

Comment:
Would be nice to get rid of that amount of duplication, as it may be hard to keep them in line if any changes are made.

I'm not exactly sure where to put common parts. We don't want the legacy 'code numbered' to depend on mixins or placeholders from new pattern. Doing it other way around and making new code snippet take styles from legacy component doesn't seem good as well.

Maybe we should have some `%code-numbered` defined in `_base_code.scss`, and extended where needed?

Reply:
I was in two minds about this - I thought about creating a placeholder, but once this pattern is removed, the only place it'll be used is in the code snippet, defeating the purpose of a placeholder. 

As we're deprecating the older pattern, and it'll ultimately get removed, I thought duplication wasn't so bad in this case and that the older pattern could potentially stay "frozen" this way until it is removed.

Reply:
While we are deprecating it, it's not like we are going to remove it soon, and we want the style consistent, so I don't think we want to keep 2 separate implementations.

Yes, using placeholder means it will technically not be needed later, when there is one component only. But it also doesn't hurt.

I wouldn't mind if it's just a bit of CSS, but there is quite a lot of very specific styling here which we may want to keep up to date properly.